### PR TITLE
fix: net8 JSON serialization

### LINKED ISSUE #2788: Default JSON TypeInfoResolver
Out of the box, Sentry.init() doesn't work for apps compiled with native AOT. The following console app fails to serialize with the error later below:
```c#
using Sentry;

// Initialize the Sentry SDK.  (It is not necessary to dispose it.)
SentrySdk.Init(options =>
{
    options.Dsn = Environment.GetCommandLineArgs()[1];
    options.Debug = true;
});

throw new ApplicationException("Something happened!");
```

```
System.InvalidOperationException: Reflection-based serialization has been disabled for this application. Either use the source generator APIs or explicitly configure the 'JsonSerializerOptions.TypeInfoResolver' property.
   at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_JsonSerializerIsReflectionDisabled() + 0x2b
   at System.Text.Json.JsonSerializerOptions.ConfigureForJsonSerializer() + 0x31
   at System.Text.Json.JsonSerializer.GetTypeInfo(JsonSerializerOptions, Type) + 0x2a
   at System.Text.Json.JsonSerializer.GetTypeInfo[T](JsonSerializerOptions) + 0x25
   at Sentry.Internal.Extensions.JsonExtensions.InternalSerializeToUtf8Bytes(Object) + 0x6a
   at Sentry.Internal.Extensions.JsonExtensions.WriteDynamicValue(Utf8JsonWriter, Object, IDiagnosticLogger) + 0x41b
   at Sentry.Internal.Extensions.JsonExtensions.WriteDynamic(Utf8JsonWriter, String, Object, IDiagnosticLogger) + 0x4b
```

### PR DESCRIPTION
**Pull request by vaind**
closes #2788

See comments in the linked issue for details (and also https://devblogs.microsoft.com/dotnet/system-text-json-in-dotnet-8/). In short, we must check `JsonSerializer.IsReflectionEnabledByDefault` instead of relying on "IsAOT" and also it's only available since net8.0 so we must add a new TF.

- fix: json serialzation on JIT net8 when PublishAot=true (requires adding a net8.0 target for Sentry)
- refactor: rename IsAot to IsNativeAot

TODO: tests (depends on #2791)

#skip-changelog - this hasn't been released yet

### CODE REVIEW

**Review by bruno-garcia — APPROVED** [review]

**Comment by Daniel15** [comment]
Is there a nightly build I can use before this fix is published to production?

**Comment by vaind** [comment]
> Is there a nightly build I can use before this fix is published to production? Trying to enable AOT for one of my apps and I hit this issue :)

There's a prerelease on nuget https://www.nuget.org/packages/Sentry/4.0.0-beta.0

File: src/Sentry/Internal/Extensions/JsonExtensions.cs [hunk 1]
```diff
@@ -551,15 +551,18 @@ private static JsonSerializerContext GetSerializerContext(Type type, bool preser
 
     private static byte[] InternalSerializeToUtf8Bytes(object value)
     {
-        return AotHelper.IsAot
-            ? AotSerializeToUtf8Bytes()
-            : JitSerializeToUtf8Bytes();
-
+#if NET8_0_OR_GREATER
```
Comment (jamescrosswell) [inline src/Sentry/Internal/Extensions/JsonExtensions.cs:554]: @vaind I think this implicitly means we need to change when we support AOT compilation:
https://github.com/getsentry/sentry-dotnet/blob/ebaed3e131d7346ce15ad28a11d37bcd30a99b0a/src/Directory.Build.props#L38-L41

If the serialization only works on net8.0 or greater then the framework only supports AOT on net8.0 or greater as well right (due to trimming)?

Comment (bruno-garcia) [reply src/Sentry/Internal/Extensions/JsonExtensions.cs:554]: Ah that's right AOT is around since .NET 7, .NET 8 added macOS support

Comment (vaind) [reply src/Sentry/Internal/Extensions/JsonExtensions.cs:554]: This condition must be for net8 because of a new API it uses.

Comment (bruno-garcia) [reply src/Sentry/Internal/Extensions/JsonExtensions.cs:554]: Good to know we have net8-only code already to validate the need to bump it.
That alone is a reason to add net8 to main packages that can be added alone (and hence bringing the net8 version of Sentry)

**Pull request merged**